File: app/src/main/java/com/libertystudio/bookshop/fragment/PurchaseFragment.java
package com.libertystudio.bookshop.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import com.libertystudio.bookshop.MainActivity;
import com.libertystudio.bookshop.R;

public class PurchaseFragment extends Fragment {
    private MainActivity mainActivity;

    public PurchaseFragment() { }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_purchase, container, false);
        initElements(view);

        return view;
    }

    private void initElements(View view) {
        mainActivity = (MainActivity) getActivity();

        TextView tvBooksCount = view.findViewById(R.id.tvBooksCount);
        TextView tvSum = view.findViewById(R.id.tvSum);

        tvBooksCount.setText(String.valueOf(mainActivity.getListBasketBooks().size()));
        tvSum.setText(mainActivity.getBasketSum() + " руб.");

        mainActivity.getListBasketBooks().clear();
    }
}

File: app/src/main/java/com/libertystudio/bookshop/fragment/PurchaseFragment.kt
package com.libertystudio.bookshop.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.libertystudio.bookshop.MainActivity
import com.libertystudio.bookshop.R
import kotlinx.android.synthetic.main.fragment_purchase.*

class PurchaseFragment : BaseFragment() {
    private var mainActivity: MainActivity? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_purchase, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        initView()
    }

    private fun initView() {
        setTitle("Успешная покупка")

        mainActivity = activity as MainActivity?

        tvPurchaseBooksCount.text = mainActivity!!.listBasketBooks.size.toString()
        tvPurchaseSum.text = mainActivity!!.basketSum.toString() + " руб."

        mainActivity!!.listBasketBooks.clear()
    }
}